src/pip/_internal/download.py

```diff
@@ -889,6 +890,19 @@ def unpack_url(
         write_delete_marker_file(location)
 
 
+def parse_content_disposition(content_disposition, default_filename):
+    # type: (str, str) -> str
+    """
+    Parse filename from content_disposition.
+    return default_filename if parsed result is empty
+    """
+    type, params = cgi.parse_header(content_disposition)
+    filename = params.get('filename')
+    if filename:
+        filename = os.path.basename(filename)
```

Comment:
I still think you should define a `sanitize_content_filename()` function in addition to what you already have, and then move most of your test cases to the unit tests of that function. That will simplify the tests because you won't have the `attachment;filename="` stuff that makes it hard to tell what slashes / characters are being passed to the inner code (e.g. after `cgi.parse_header()` is called), and whether you're really covering the cases you want to test.